src/source/Sctp/Sctp.c

```diff
@@ -169,16 +169,31 @@ STATUS sctpSessionWriteMessage(PSctpSession pSctpSession, UINT32 streamId, BOOL
 {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
-    struct sctp_sendv_spa spa;
 
     CHK(pSctpSession != NULL && pMessage != NULL, STATUS_NULL_ARG);
 
-    MEMSET(&spa, 0x00, SIZEOF(struct sctp_sendv_spa));
-    spa.sendv_flags |= SCTP_SEND_SNDINFO_VALID;
-    spa.sendv_sndinfo.snd_sid = streamId;
+    MEMSET(&pSctpSession->spa, 0x00, SIZEOF(struct sctp_sendv_spa));
 
-    putInt32((PINT32) &spa.sendv_sndinfo.snd_ppid, isBinary ? SCTP_PPID_BINARY : SCTP_PPID_STRING);
-    CHK(usrsctp_sendv(pSctpSession->socket, pMessage, pMessageLen, NULL, 0, &spa, SIZEOF(spa), SCTP_SENDV_SPA, 0) > 0, STATUS_INTERNAL_ERROR);
+    pSctpSession->spa.sendv_flags |= SCTP_SEND_SNDINFO_VALID;
+    pSctpSession->spa.sendv_sndinfo.snd_sid = streamId;
+
+    if ((pSctpSession->packet[1] & DCEP_DATA_CHANNEL_RELIABLE_UNORDERED) != 0) {
+        pSctpSession->spa.sendv_sndinfo.snd_flags |= SCTP_UNORDERED;
+    }
+    if ((pSctpSession->packet[1] & DCEP_DATA_CHANNEL_REXMIT) != 0) {
+        pSctpSession->spa.sendv_prinfo.pr_policy = SCTP_PR_SCTP_RTX;
+        MEMCPY(&pSctpSession->spa.sendv_prinfo.pr_value, pSctpSession->packet + SIZEOF(UINT32), SIZEOF(UINT32));
```

Comment:
Not sure why are you doing these? Why not simply putUnalignedInt32BitEndian directly into the destination?

Reply:
I wanted to copy between two memory locations and then convert to big endian. But, you are right. I do not need both. I replaced it with `getUnaligned32BitBigEndian` to do the same.